From ffe77aedcf378b95f45d51ec65d313156c957343 Mon Sep 17 00:00:00 2001
From: Kai Vehmanen <kai.vehmanen@linux.intel.com>
Date: Wed, 10 Jun 2020 17:14:16 +0300
Subject: [PATCH 05/14] sof-hda-dsp: don't fail if Auto-Mute control is not
 present

The "Auto-Mute Mode" control is not present in all HDA codecs.
The generic SOF HDA UCM file should be robust enough to handle
these cases as well.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
(cherry picked from commit 4c1189fdcb4819df6c36cf490d17f532f82c8bbe)
Signed-off-by: Hui Wang <hui.wang@canonical.com>
---
 ucm2/sof-hda-dsp/HiFi.conf | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/ucm2/sof-hda-dsp/HiFi.conf b/ucm2/sof-hda-dsp/HiFi.conf
index 9a82930..eea6ed9 100644
--- a/ucm2/sof-hda-dsp/HiFi.conf
+++ b/ucm2/sof-hda-dsp/HiFi.conf
@@ -1,9 +1,17 @@
 # Use case Configuration for sof-hda-dsp
 
 SectionVerb {
-	EnableSequence [
-		cset "name='Auto-Mute Mode' 'Disabled'"
-	]
+	If.automute {
+		Condition {
+			Type ControlExists
+			Control "name='Auto-Mute Mode'"
+		}
+		True {
+			EnableSequence [
+				cset "name='Auto-Mute Mode' 'Disabled'"
+			]
+		}
+	}
 }
 
 SectionDevice."Headphones" {
-- 
2.17.1

